### PR TITLE
Fix inappropriate non-modal drawer closure

### LINKED ISSUE #331: drawer with modal=false still closes inappropriately
Though #109  fixed the closing issue for general mouse clicks, it did not extend the fix to two other cases:

1) When the user clicks on a focusable element (button, input, etc)
2) When the user presses the ESC key.

Both of these still close the drawer, even if modal=false

This is specifically referenced in #37, and #114 
This may also affect #184 and others 

I'm attaching a PR to this.

### PR DESCRIPTION
**Pull request by artemis-prime**
fixes #331 and follows on #109
added preventDefault to onFocusOutside() and onEscapeKeyDown()
website: added 'test-modal' route with demo of issue / fix

Without the added code, clicking on the 'Does nothing' button, (or any focasable element) will close the drawer, even if non-modal.
Without the added code, hitting the ESC key will close the drawer, even if non-modal. 

tl;dr:

#109 handled one of case that closes the drawer by default, but not the other two:
 
https://github.com/radix-ui/primitives/blob/main/packages/react/dismissable-layer/src/DismissableLayer.tsx
```typescript
    const pointerDownOutside = usePointerDownOutside((event) => {
      const target = event.target as HTMLElement;
      const isPointerDownOnBranch = [...context.branches].some((branch) => branch.contains(target));
      if (!isPointerEventsEnabled || isPointerDownOnBranch) return;
      onPointerDownOutside?.(event);
      onInteractOutside?.(event);
      if (!event.defaultPrevented) onDismiss?.();
    }, ownerDocument);

    const focusOutside = useFocusOutside((event) => {
      const target = event.target as HTMLElement;
      const isFocusInBranch = [...context.branches].some((branch) => branch.contains(target));
      if (isFocusInBranch) return;
      onFocusOutside?.(event);
      onInteractOutside?.(event);
      if (!event.defaultPrevented) onDismiss?.();
    }, ownerDocument);

    useEscapeKeydown((event) => {
      const isHighestLayer = index === context.layers.size - 1;
      if (!isHighestLayer) return;
      onEscapeKeyDown?.(event);
      if (!event.defaultPrevented && onDismiss) {
        event.preventDefault();
        onDismiss();
      }
    }, ownerDocument);

### CODE REVIEW

**Comment by franzwollang** [comment]
I would appreciate this fix. Having problems with this exact issue.
